crates/library/std/src/evm.fe

```diff
@@ -0,0 +1,293 @@
+# Basic context accessor functions.
+# These should be unsafe when the Context object is introduced.
+
+pub fn chain_id() -> u256:
+  unsafe:
+    return __chainid()
+
+pub fn base_fee() -> u256:
+  unsafe:
+    return __basefee()
+
+pub fn origin() -> address:
+  unsafe:
+    return address(__origin())
+
+pub fn gas_price() -> u256:
+  unsafe:
+    return __gasprice()
+
+pub fn block_hash(b: u256) -> u256:
+  unsafe:
+    return __blockhash(b)
+
+pub fn coin_base() -> address:
+  unsafe:
+    return address(__coinbase())
+
+pub fn timestamp() -> u256:
+  unsafe:
+    return __timestamp()
+
+pub fn block_number() -> u256:
+  unsafe:
+    return __number()
+
+pub fn difficulty() -> u256:
+  unsafe:
+    return __difficulty()
+
+pub fn gas_limit() -> u256:
+  unsafe:
+    return __gaslimit()
+
+pub fn gas() -> u256:
+  unsafe:
+    return __gas()
+
+pub fn self_address() -> address:
+  unsafe:
+    return address(__address())
+
+pub fn balance_of(addr: address) -> u256:
+  unsafe:
+    return __balance(u256(addr))
+
+pub fn balance() -> u256:
+  unsafe:
+    return __selfbalance()
+
+pub fn caller() -> address:
+  unsafe:
+    return address(__caller())
+
+pub fn call_value() -> u256:
+  unsafe:
+    return __callvalue()
+
+
+# Overflowing math ops. Should these be unsafe or named
+# `overflowing_add`, etc?
```

Comment:
Without taking very long to think about it I'm in favor of making these `unsafe`

Reply:
`unsafe` feels a little strong to me for arithmetic; like perhaps shouldn't "weaken" it by including wrapping ops in that category. I ended up just leaving it as-is; anyone with strong feelings is welcome to change these of course. Btw, we had a brief discussion on discord while you weren't around to stop us about making the operations on the base types wrap without revert (gasp!), and introducing `SafeU32` etc types that `impl Add` with revert-on-overflow behavior. A discussion to rehash at some point.